### PR TITLE
Release 3.4.1

### PR DESCRIPTION
**Pull request by ankush-sqops**
Updated terraform-aws-vpc module versions:

- Updated module versions.
- Updated variables
- Updated VPN storage class and root volume encryption
- Updated karpenter private subnet tag

### CODE REVIEW

File: modules/vpn/main.tf [hunk 1]
```diff
@@ -59,7 +61,8 @@ module "security_group_vpn" {
   )
 }
 
-data "aws_ami" "ubuntu_20_ami" {
```
Comment (anish-squareops) [inline modules/vpn/main.tf:62]: @ankush-sqops Can we hardcode the AMI id here. 
Reason: When we reapply the module after a month or so, the VPN instance gets terminated and new instance comes up with new base AMI as AWS launches new AMI every now and then.

Comment (ankush-sqops) [reply modules/vpn/main.tf:62]: Okay, I'll check and try to hardcode the AMI

Comment (ankush-sqops) [reply modules/vpn/main.tf:62]: @anish-squareops  Updated the PR, please check the changes.

**Comment by vinayakgautamops** [comment]
image


tag value is wrong according to the test case 
Currently passing value:
- Karpenter: vpc-us-east-1a

expected value:
- Karpenter: us-east-1a

@ankush-sqops

**Comment by vinayakgautamops** [comment]
AWS recommends storing VPC Flow Logs for retention periods based on security and compliance requirements. For most use cases, a retention period of 30 days is sufficient. This duration provides a balance between data storage and the need to retain flow logs for security monitoring and auditing purposes.

@ankush-sqops Please update `flow_log_cloudwatch_log_group_retention_in_days` from` 90 to 30` days 
@yuvraj-squareops1 let us know your thoughts on this

**Comment by vinayakgautamops** [comment]
@ankush-sqops @anish-squareops @yuvraj-squareops1 There should be an option for select all az's by one variable.
image

**Pull request merged**